[
    {
        "type": "message",
        "user": "U1TA5FDU7",
        "text": "Hi, everyone. Asking your help. When i test fabric-v1.0, I got this error, can anyone give me some clue?\n```\nunit-tests_1 | --- FAIL: TestConfigerInvokeJoinChainCorrectParams (0.03s)\nunit-tests_1 |  configer_test.go:55: Local peer IP address: 172.17.0.3:21213\nunit-tests_1 | panic: ---empty version---(chain=mytestchainid,chaincode=lccc,version=,txid=9f5c8a3d-a738-4054-8f27-cdc3bc094bda,syscc=true,proposal=0x0 [recovered]\nunit-tests_1 |  panic: ---empty version---(chain=mytestchainid,chaincode=lccc,version=,txid=9f5c8a3d-a738-4054-8f27-cdc3bc094bda,syscc=true,proposal=0x0\nunit-tests_1 |\nunit-tests_1 | goroutine 23 [running]:\nunit-tests_1 | panic(0xa4aa60, 0xc4202e7d00)\nunit-tests_1 |  \/opt\/go\/src\/runtime\/panic.go:500 +0x1a1\nunit-tests_1 | testing.tRunner.func1(0xc420283c80)\nunit-tests_1 |  \/opt\/go\/src\/testing\/testing.go:579 +0x25d\nunit-tests_1 | panic(0xa4aa60, 0xc4202e7d00)\nunit-tests_1 |  \/opt\/go\/src\/runtime\/panic.go:458 +0x243\nunit-tests_1 | <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode.NewCCContext|github.com\/hyperledger\/fabric\/core\/chaincode.NewCCContext>(0xc4202e77e0, 0xd, 0xb50520, 0x4, 0x0, 0x0, 0xc4202e2510, 0x24, 0xabc901, 0x0, ...)\nunit-tests_1 |  <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/chaincode_support.go:87|github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/chaincode_support.go:87> +0xa43\nunit-tests_1 | <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode.deploySysCC(0xc4202e77e0|github.com\/hyperledger\/fabric\/core\/chaincode.deploySysCC(0xc4202e77e0>, 0xd, 0xec06c0, 0x0, 0x0)\nunit-tests_1 |  <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/sysccapi.go:130|github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/sysccapi.go:130> +0x59f\nunit-tests_1 | <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode.DeploySysCCs(0xc4202e77e0|github.com\/hyperledger\/fabric\/core\/chaincode.DeploySysCCs(0xc4202e77e0>, 0xd)\nunit-tests_1 |  <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/importsysccs.go:62|github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/importsysccs.go:62> +0xab\nunit-tests_1 | <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode.joinChain(0xc4202f3200|github.com\/hyperledger\/fabric\/core\/chaincode.joinChain(0xc4202f3200>, 0x11ca, 0x1200, 0x9, 0x1, 0x1, 0x410f2e, 0xc4202cbc80)\nunit-tests_1 |  <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/configer.go:119|github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/configer.go:119> +0x269\nunit-tests_1 | <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode.(*PeerConfiger).Invoke(0xee6bb8|github.com\/hyperledger\/fabric\/core\/chaincode.(*PeerConfiger).Invoke(0xee6bb8>, 0xea3800, 0xc4202cc180, 0x411758, 0x30, 0xa54c00, 0x1, 0xc4202cbc80)\nunit-tests_1 |  <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/configer.go:69|github.com\/hyperledger\/fabric\/core\/chaincode\/_test\/_obj_test\/configer.go:69> +0x544\nunit-tests_1 | <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode\/shim.(*MockStub).MockInvoke(0xc4202cc180|github.com\/hyperledger\/fabric\/core\/chaincode\/shim.(*MockStub).MockInvoke(0xc4202cc180>, 0xb4f909, 0x1, 0xc4202cbc80, 0x2, 0x2, 0x0, 0x1, 0x0, 0xc4202ca870, ...)\nunit-tests_1 |  \/opt\/gopath\/src\/github.com\/hyperledger\/fabric\/core\/chaincode\/shim\/mockstub.go:120 +0x95\nunit-tests_1 | <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode.TestConfigerInvokeJoinChainCorrectParams(0xc420283c80)|github.com\/hyperledger\/fabric\/core\/chaincode.TestConfigerInvokeJoinChainCorrectParams(0xc420283c80)>\nunit-tests_1 |  \/opt\/gopath\/src\/github.com\/hyperledger\/fabric\/core\/chaincode\/configer_test.go:125 +0x51c\nunit-tests_1 | testing.tRunner(0xc420283c80, 0xbba550)\nunit-tests_1 |  \/opt\/go\/src\/testing\/testing.go:610 +0x81\nunit-tests_1 | created by testing.(*T).Run\nunit-tests_1 |  \/opt\/go\/src\/testing\/testing.go:646 +0x2ec\nunit-tests_1 | FAIL     <http:\/\/github.com\/hyperledger\/fabric\/core\/chaincode|github.com\/hyperledger\/fabric\/core\/chaincode>    0.081s\nunit-tests_1 | error: exit status 1\nunit-tests_1 | panic: EOF\n```",
        "ts": "1484296506.001598"
    },
    {
        "type": "message",
        "user": "U1TA5FDU7",
        "text": "It said \"empty version\".  How can i fix this.",
        "ts": "1484296562.001599"
    },
    {
        "type": "message",
        "user": "U1TA5FDU7",
        "text": "Also, when I execute ```docker-compose -f docker-compose-with-orderer.yml up``` in fabric\/bdtests\/. It returns \n```ERROR: In file '.\/docker-compose-with-orderer.yml' service 'version' doesn't have any configuration options. All top level keys in your docker-compose.yml ns.```",
        "edited": {
            "user": "U1TA5FDU7",
            "ts": "1484299455.000000"
        },
        "ts": "1484299408.001600"
    },
    {
        "type": "message",
        "user": "U12452RAP",
        "text": "seems you should upgrade your docker compose ",
        "ts": "1484303956.001602",
        "reactions": [
            {
                "name": "+1",
                "users": [
                    "U1TA5FDU7"
                ],
                "count": 1
            }
        ]
    },
    {
        "type": "message",
        "user": "U37GEHJSV",
        "text": "Does anyone have an example which demonstrates how to send events from blockchain to, for example, a nodeJS application?",
        "ts": "1484304455.001603"
    },
    {
        "user": "U3LF97LF3",
        "text": "<@U3LF97LF3|sneha> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1484312222.001604"
    },
    {
        "type": "message",
        "user": "U3LF97LF3",
        "text": "Hi All,\nWe are facing issue while invoking chaincode on fabric v1.0.  Could you please provide your inputs. \nRegistration and deployment of chaincode is successful. However when we try to invoke chaincode using CLI we are getting error.\nCommand:- peer chaincode invoke -n mycc -c '{\"Args\":[\"write\",\"hello_world\",\"200\"]}' \n\n12:42:05.880 [main] main -&gt; INFO 001 User defined config file path: \/etc\/hyperledger\/fabric\n12:42:05.883 [msp] func1 -&gt; INFO 002 Creating the msp manager\n12:42:05.883 [msp] GetManager -&gt; INFO 003 Returning MSP manager %!p(msp.peerMspManagerImpl={map[]  false})\n12:42:05.883 [msp] Setup -&gt; INFO 004 Setting up the MSP manager from config file \/etc\/hyperledger\/fabric\/msp\/peer-config.json\n12:42:05.883 [msp] newBccspMsp -&gt; INFO 005 Creating BCCSP-based MSP instance\n12:42:05.884 [SW_BCCSP] init -&gt; WARN 006 'security.bccsp.default.keyStorePath' not set. Using the default directory [%s] for temporary files \/tmp\n12:42:05.884 [SW_BCCSP] init -&gt; INFO 007 Root Path [\/tmp]\n12:42:05.884 [SW_BCCSP] createKeyStoreIfNotExists -&gt; INFO 008 Keystore path [\/tmp\/crypto\/ks] missing [true]: [&lt;clean&gt;]\n12:42:05.884 [msp] Setup -&gt; INFO 009 Setting up MSP DEFAULT\n12:42:05.884 [msp] Setup -&gt; INFO 00a Setting up MSP instance from file \/etc\/hyperledger\/fabric\/msp\/peer-config.json\n12:42:05.909 [msp] newIdentity -&gt; INFO 00b Creating identity instance for ID &amp;{{DEFAULT} ROOTCA}\n12:42:05.909 [msp] newSigningIdentity -&gt; INFO 00c Creating signing identity instance for ID &amp;{{DEFAULT} PEER}\n12:42:05.909 [msp] Setup -&gt; INFO 00d MSP manager setup complete (config file \/etc\/hyperledger\/fabric\/msp\/peer-config.json)\n12:42:05.909 [logging] LoggingInit -&gt; DEBU 00e Setting default logging level to DEBUG for command 'chaincode'\n12:42:05.912 [msp] GetManager -&gt; INFO 00f Returning MSP manager %!p(msp.peerMspManagerImpl={map[DEFAULT:0xc4201f8570] PeerMSPManager true})\n12:42:05.912 [msp] GetSigningIdentity -&gt; INFO 010 Looking up MSP with ID {DEFAULT}\n12:42:05.912 [msp] GetSigningIdentity -&gt; INFO 011 Obtaining signing identity for &amp;{{DEFAULT} PEER}\n12:42:05.914 [msp] Sign -&gt; INFO 012 Signing message\nError: Error endorsing chaincode: rpc error: code = 2 desc = Error deploying chaincode: Failed to launch chaincode spec(Failed to init chaincode(handler not found for chaincode mycc))",
        "ts": "1484313456.001605"
    },
    {
        "type": "message",
        "user": "U0ULK2JPP",
        "text": "<@U3LF97LF3> can you share your steps for testing above please ?",
        "ts": "1484315071.001606"
    },
    {
        "type": "message",
        "user": "U0ULK2JPP",
        "text": "1) how the  orderer is started (2) how is the peer started",
        "ts": "1484315111.001607"
    },
    {
        "type": "message",
        "user": "U0ULK2JPP",
        "text": "<@U1TA5FDU7> what command are you using to run the tests ?",
        "edited": {
            "user": "U0ULK2JPP",
            "ts": "1484315210.000000"
        },
        "ts": "1484315184.001608"
    },
    {
        "user": "U3PC934K0",
        "text": "<@U3PC934K0|erickan> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1484322403.001611"
    },
    {
        "user": "U1P56CWF5",
        "text": "<@U1P56CWF5|yashgt> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1484327010.001612"
    },
    {
        "user": "U0UHG4VP1",
        "text": "<@U0UHG4VP1|ry> has left the channel",
        "type": "message",
        "subtype": "channel_leave",
        "ts": "1484330815.001615"
    },
    {
        "user": "U3KHE03PS",
        "text": "<@U3KHE03PS|tooraj.musk> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1484338953.001616"
    },
    {
        "user": "U2Y9JL8UU",
        "text": "<@U2Y9JL8UU|crmiles> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1484339325.001617"
    },
    {
        "type": "message",
        "user": "U3DKG7M54",
        "text": "<@U0ULK2JPP>, <@U0PB67X4K>   I was looking at the same issue and working with <@U3LF97LF3>. There is gRPC error from orderer when it tries to send the Deploy Transaction Block to Peer. Port exposed by peer by default is 7051. However orderer uses the right docker container IP of Commiter Peer (which is also endorser for 1 Peer Setup) but a different port (48144 and randomly in increasing order). This results in write broken pipe error. How does Orderer know about the list of Committer Peers (IP, Port) on the Hyperledger network, is there a way can provide more debug information or is there a known solution?\n\nSetup: Fabric {peer, cop and orderer} images are used from the connect-a-thon - v1.0 architecture and is currently used to run a custom-helloworld-chaincode  using DEV mode.\n\nSecondly - I have tried this on Linux box hosted on AWS and has never seen this problem however on our internal organization boxes (Intranet one of the environment difference), we see this error almost frequently.",
        "edited": {
            "user": "U3DKG7M54",
            "ts": "1484403843.000000"
        },
        "ts": "1484348167.001620"
    }
]